src/routes/snippets.rs

```diff
@@ -61,6 +61,7 @@ pub fn create_snippet(
     origin: &Origin,
     body: Result<Input<NewSnippet>, ApiError>,
     _content_type: NegotiatedContentType,
+    _role: BearerAuth,
```

Comment:
It does not. The way `BearerAuth` works is that it will try to authenticate the user if a valid token is passed in the `Authorization` header, and will resolve to `Role::Guest` if nothing is passed.